### PR TITLE
[docathon] Clean up metadata docs (DOC-382)

### PR DESCRIPTION
**Pull request by petehunt**
## Summary & Motivation

I learned some new things during @erinkcochran87's presentation

## Changelog [New | Bug | Docs]

NOCHANGELOG

### CODE REVIEW

File: docs/docs-beta/docs/guides/data-modeling/metadata.md [hunk 1]
```diff
@@ -30,30 +30,34 @@ To follow the steps in this guide, you'll need:
...
+With Dagster+ Pro, you can create asset-based alerts that will automatically notify an asset's owners when triggered. Refer to the [Dagster+ alert documentation](/dagster-plus/deployment/alerts) for more information.
 
+:::
 
 ## Choosing between tags or metadata for custom information
 
 In Dagster, you can attach custom information to assets in two ways: **tags** and **metadata**.
 
-**Tags** are a simple way to organize assets in Dagster. You can attach several tags to an asset when it is defined, and they will appear in the UI. You can also use tags to search and filter for assets in the [asset catalog](/todo). They are structured as key-value pairs of strings.
+**Tags** are the primary way to organize assets in Dagster. You can attach several tags to an asset when it's defined, and they will appear in the UI. You can also use tags to search and filter for assets in the [Asset catalog](/todo). They're structured as key-value pairs of strings.
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/guides/data-modeling/metadata.md:49]: It's a thing of beauty

File: docs/docs-beta/docs/guides/data-modeling/metadata.md [hunk 2]
```diff
@@ -30,30 +30,34 @@ To follow the steps in this guide, you'll need:
...
 
 Here's an example of some tags one might apply to an asset:
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/guides/data-modeling/metadata.md:51]: Nit - can we switch `one` to `you`, even though it's technically not part of this PR? It's giving me college essay vibes

File: docs/docs-beta/docs/guides/data-modeling/metadata.md [hunk 3]
```diff
@@ -30,30 +30,34 @@ To follow the steps in this guide, you'll need:
...
+
 ```python
 {"domain": "marketing", "pii": "true"}
 ```
 
 **Metadata** allows you to attach rich information to the asset, like a Markdown description, a table schema, or a time series. Metadata is more flexible than tags, as it can store more complex information. Metadata can be attached to an asset at definition time (i.e. when the code is first imported) or at runtime (every time an asset is materialized).
 
 Here's an example of some metadata one might apply to an asset:
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/guides/data-modeling/metadata.md:59]: Same here, re: `one` > `you`

File: docs/docs-beta/docs/guides/data-modeling/metadata.md (hunk 2 above)
Comment (erinkcochran87) [inline docs/docs-beta/docs/guides/data-modeling/metadata.md:51]: Nit - can we switch `one` to `you`, even though it's technically not part of this PR? It's giving me college essay vibes

File: docs/docs-beta/docs/guides/data-modeling/metadata.md [hunk 4]
```diff
@@ -114,22 +116,33 @@ Here's a quick example of how to attach this metadata at both definition time an
...
 <CodeExample filePath="guides/data-modeling/metadata/table-column-lineage-metadata.py" language="python" title="Table column lineage metadata" />
 
-> Dagster+ provides rich visualization and navigation of column lineage in the asset catalog. Refer to the [Dagster+ documentation](/dagster-plus) for more information.
+:::tip
+
+Dagster+ provides rich visualization and navigation of column lineage in the asset catalog. Refer to the [Dagster+ documentation](/dagster-plus) for more information.
 
+:::
 
 ## Linking your assets with their source code
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/guides/data-modeling/metadata.md:139]: Nit - let's cut `your` out of this to make the title shorter.

File: docs/docs-beta/docs/guides/data-modeling/metadata.md [hunk 5]
```diff
@@ -143,13 +156,13 @@ Dagster can automatically attach code references to your assets during local dev
 
 ### Attaching custom code references for local development
 
-You can manually add the `dagster/code_references` metadata to your asset definitions to attach custom code references. This is useful when your asset's source code is not primarily in Python (e.g. when parsing a YAML config or integrating with another language).
+You can manually add the `dagster/code_references` metadata to your asset definitions to attach custom code references. This is useful when your asset's source code isn't primarily in Python (e.g. when parsing a YAML config or integrating with another language).
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/guides/data-modeling/metadata.md:159]: A few thoughts here:

- Should this be reframed to `Attaching non-Python code references`? I looked at this and wasn't sure what `custom` meant until I finished reading the intro paragraph
- Let's switch the sentences in this paragraph around so the use case comes before the solution. For example, something like:

   > If an asset's source code isn't primarily in Python - such as parsing a YAML config or integrating with another language - you can still attach code references. Manually attaching the `dagster/code_references` metadata to your asset definitions will allow you to link to the asset's source code.

File: docs/docs-beta/docs/guides/data-modeling/metadata.md [hunk 6]
```diff
@@ -143,13 +156,13 @@ Dagster can automatically attach code references to your assets during local dev
...
 
 <CodeExample filePath="guides/data-modeling/metadata/custom-local-references.py" language="python" title="Custom local source code references" />
 
 ### Attaching code references in production (Dagster+)
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/guides/data-modeling/metadata.md:163]: Perhaps combine these sections and put `Dagster+` and `OSS` into tabs? Let me know if you'd prefer I do this as a follow up - just throwing it out there since you've been stoked to work on these :)

Screenshot 2024-08-26 at 1 54 02 PM

Code:

```
### Attaching code references in production

<Tabs>
  <TabItem value="dagster-plus" label="Dagster+">

Dagster+ can automatically annotate your assets with code references to source control such as GitHub or GitLab.

<CodeExample filePath="guides/data-modeling/metadata/plus-references.py" language="python" title="Production source code references (Dagster+)" />

</TabItem>
<TabItem value="dagster-open-source" label="OSS">

If you aren't using Dagster+, you can annotate your assets with code references to source control, but it requires a little more manual mapping.

<CodeExample filePath="guides/data-modeling/metadata/oss-references.py" language="python" title="Production source code references (OSS)" />

[`link_code_references_to_git`](/todo) currently supports GitHub and GitLab repositories. It also supports customization of how file paths are mapped; see the [`AnchorBasedFilePathMapping`](/todo) API docs for more information.

</TabItem>
</Tabs>
```

**Review by erinkcochran87 — APPROVED** [review]
This looks great! I left a few comments about things that aren't explicitly on this PR, as I didn't get a chance to review your original guide. Let me know if you'd prefer I handle anything as a follow up.

**Comment by erinkcochran87** [comment]
Looks good to me!

**Review by PedramNavid — APPROVED** [review]
simply beautiful
